plugins/org.jkiss.dbeaver.core/src/org/jkiss/dbeaver/ui/dialogs/connection/ConnectionWizard.java

```diff
@@ -234,6 +235,12 @@ public void testConnection() {
                     }
                 });
 
+                var oldUserPassword = activeDataSource.getActualConnectionConfiguration().getUserPassword();
+                var newUserPassword = targetDataSource.getActualConnectionConfiguration().getUserPassword();
+                if (newUserPassword != null && !newUserPassword.equals(oldUserPassword)) {
+                    DBUtils.fireObjectUpdate(activeDataSource, targetDataSource.getActualConnectionConfiguration());
```

Comment:
Shouldn't we add `activeDataSource == targetDataSource` condition here? I'm not sure, just asking, because now it looks like it's executed almost each time Test connection is called. I see you've added additional check to event handler, but maybe we can check it earlier also?